massa-module-cache/src/controller.rs

```diff
@@ -122,12 +116,22 @@ impl ModuleCache {
         }
     }
 
-    /// Load a cached module for execution and check its validity for execution
+    /// Load a cached module for execution and check its validity for execution.
+    /// Also checks that the provided execution gas is enough to pay for the instance creation cost.
+    ///
+    /// Returns the module and the remaining gas after loading.
     pub fn load_module(
         &mut self,
         bytecode: &[u8],
         execution_gas: u64,
-    ) -> Result<RuntimeModule, CacheError> {
+    ) -> Result<(RuntimeModule, u64), CacheError> {
+        // TODO: interesting but unimportant optim
+        // remove max_instance_cost hard check if module is cached and has a delta
+        execution_gas
```

Comment:
same as below
should probably be `execution_gas = execution_gas.checked_sub()` else it seems the gas is not paid